### PR TITLE
scripts: follow up of #3115 (feedback)

### PR DESCRIPTION
**Pull request by georglauterbach**
# Description

Follow-up of #3115 (branch name of this PR wrong, lol). See
<https://github.com/docker-mailserver/docker-mailserver/pull/3115#discussion_r1118077293> for example.

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: target/scripts/start-mailserver.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+shopt -s globstar
```
Comment (casperklein) [inline target/scripts/start-mailserver.sh:3]: Can this introduce unwanted side-effects with our existing scripts that are called afterwards?

Another option would be, to just use it in the `_setup` function.

Comment (georglauterbach) [reply target/scripts/start-mailserver.sh:3]: I was wondering about this too. But I left it up to the tests. Seems there is one place we need to adjust. I will do this later.

Comment (georglauterbach) [reply target/scripts/start-mailserver.sh:3]: Nevermind, was a sporadic BATS issue.

Comment (polarathene) [reply target/scripts/start-mailserver.sh:3]: Probably does make sense to move it into `_setup` if that is where this is only relevant? I believe we're using it somewhere else for change detection in a similar fashion.

**Review by polarathene — APPROVED** [review]

File: target/scripts/start-mailserver.sh (hunk 1 above)
Comment (georglauterbach) [reply target/scripts/start-mailserver.sh:3]: I decided against this for two reasons:

1. The effects of `shopt` are not scoped to a function, i.e.

```console
$ cat test.sh
#!/bin/bash

shopt -u globstar
shopt | grep "globstar"

function a() {
  shopt -s globstar
  shopt | grep "globstar"
}

a
shopt | grep "globstar"
$ bash test.sh
globstar     off
globstar     on
globstar     on
```

Hence, setting this inside the whole setup, i.e. changing the behavior at a certain point, is undesirable. We could turn it off again, but actually setting it as fine I think; we may be able to use it in the future and it does not hurt at all.

2. I want Bash adjustments like `shopt` or `set` to be used at the very beginning of scripts so everyone immediately notices them. I plan on introducing `set -u` and `shopt -s inherit_errexit` too, and these should go together at the beginning `start-mailserver.sh`.

**Review by casperklein — APPROVED** [review]

File: target/scripts/start-mailserver.sh (hunk 1 above)
Comment (casperklein) [reply target/scripts/start-mailserver.sh:3]: I am aware of that. I didn't express it clearly enough. I meant something like this:

function foo() {
  shopt -s globstar
  for i in /foo/**/*; do
		bar
  done
  shopt -u globstar
}

Setting globstar in the function and revert it after usage.

Comment (georglauterbach) [reply target/scripts/start-mailserver.sh:3]: I actually think that this feature is useful in general, and having it available everywhere is a nice thing :)

**Pull request merged**